project/DistributionPackage.scala

```diff
@@ -419,14 +425,23 @@ object DistributionPackage {
     }
   }
 
-  /** Returns the index of the next argument after `--run`, if it exists. */
-  private def locateRunArgument(args: Seq[String]): Option[Int] = {
-    val findRun = args.indexOf("--run")
-    if (findRun >= 0 && findRun + 1 < args.size) {
-      Some(findRun + 1)
-    } else {
-      None
+  /** Returns the argument specifying the path of the project to run.
+    *
+    * It will be the argument following `--in-project`, `--run` or `--compile`.
+    */
+  private def findProjectPath(args: Seq[String]): Option[String] = {
```

Comment:
Note that once #12906 is integrated (should be soon), the `--in-project` cmdline option will be kind of obsolete.

Reply:
Do you mean you are removing `--in-project` option there?

Then, once merged we can remove it from here too.